x-pack/plugins/dataset_quality/public/state_machines/summary_panel/src/state_machine.ts

```diff
@@ -0,0 +1,241 @@
+      predictableActionArguments: true,
+      id: 'DatasetsQualitySummaryPanel',
+      type: 'parallel',
+      states: {
+        datasetsQuality: {
+          initial: 'fetching',
+          states: {
+            fetching: {
+              invoke: {
+                src: 'loadDatasetsQuality',
+                onDone: {
+                  target: 'loaded',
+                  actions: ['storeDatasetsQuality'],
+                },
+                onError: [
+                  {
+                    target: 'retrying',
+                    cond: {
+                      type: 'canRetry',
+                      counter: 'datasetsQualityRetries',
+                    },
+                    actions: ['incrementDatasetsQualityRetries'],
+                  },
+                  {
+                    target: 'loaded',
+                    actions: ['notifyFetchDatasetsQualityFailed'],
+                  },
+                ],
+              },
+            },
+            retrying: {
+              after: {
+                [RETRY_DELAY_IN_MS]: 'fetching',
+              },
+            },
+            loaded: {
+              type: 'final',
+            },
+          },
+        },
+        datasetsActivity: {
+          initial: 'fetching',
+          states: {
+            fetching: {
+              invoke: {
+                src: 'loadDatasetsActivity',
+                onDone: {
+                  target: 'loaded',
+                  actions: ['storeDatasetsActivity'],
+                },
+                onError: [
+                  {
+                    target: 'retrying',
+                    cond: {
+                      type: 'canRetry',
+                      counter: 'datasetsActivityRetries',
+                    },
+                    actions: ['incrementDatasetsActivityRetries'],
+                  },
+                  {
+                    target: 'loaded',
+                    actions: ['notifyFetchDatasetsActivityFailed'],
+                  },
+                ],
+              },
+            },
+            retrying: {
+              after: {
+                [RETRY_DELAY_IN_MS]: 'fetching',
+              },
+            },
+            loaded: {
+              type: 'final',
+            },
+          },
+        },
+        estimatedData: {
+          initial: 'fetching',
+          states: {
+            fetching: {
+              invoke: {
+                src: 'loadEstimatedData',
+                onDone: {
+                  target: 'loaded',
+                  actions: ['storeEstimatedData'],
+                },
+                onError: [
+                  {
+                    target: 'retrying',
+                    cond: {
+                      type: 'canRetry',
+                      counter: 'estimatedDataRetries',
+                    },
+                    actions: ['incrementEstimatedDataRetries'],
+                  },
+                  {
+                    target: 'loaded',
```

Comment:
How do this works? we only enter in the second object of the errors if we cannot retry or always?

Reply:
TIL 😄